yoga/Yoga.h

```diff
@@ -364,6 +374,20 @@ YG_EXTERN_C_END
 #include <functional>
 #include <vector>
 
+// Templated delegates for YGMemoryAllocate & YGMemoryFree, so we don't have to
+// cast nor pass in the size of the allocated chunk of memory explicitly.
+template<typename T, typename... A>
+T* YGAllocate(A&&... arguments) {
+  auto* memory = reinterpret_cast<T*>(YGMemoryAllocate(sizeof(T), alignof(T)));
+  new(memory) T(std::forward<A>(arguments)...);
+  return memory;
+}
+template<typename T>
+void YGFree(T* memory) {
+  memory->~T();
+  YGMemoryFree(memory);
+}
+
```

Comment:
This should not be part of the public API, since it is a helper for the internals of Yoga.